contrib/rocksdb/template.py

```diff
@@ -0,0 +1,58 @@
+pkgname = "rocksdb"
+pkgver = "9.4.0"
+pkgrel = 0
+build_style = "makefile"
+make_cmd = "gmake"
+make_build_target = "static_lib"
+hostmakedepends = [
+    "bash",
+    "gmake",
+    "perl",
+    "pkgconf",
+]
+makedepends = [
+    "bzip2-devel",
+    "gflags-devel-static",
+    "linux-headers",
+    "lz4-devel",
+    "snappy-devel",
+    "zstd-devel",
+    "zlib-ng-compat-devel",
+]
+pkgdesc = "Embeddable, persistent key-value store for fast storage"
+maintainer = "Gnarwhal <git.aspect893@passmail.net>"
+license = "GPL-2.0-or-later OR Apache-2.0 OR custom:leveldb"
+url = "https://rocksdb.org"
+source = f"https://github.com/facebook/rocksdb/archive/refs/tags/v{pkgver}.tar.gz"
+sha256 = "1f829976aa24b8ba432e156f52c9e0f0bd89c46dc0cc5a9a628ea70571c1551c"
+hardening = ["!pie"]
+
+
+def do_build(self):
+    # by default, pass various stuff directly rather than through env
+    tool_args = [
+        "PREFIX=/usr",
+        "PKG_CONFIG=" + self.get_tool("PKG_CONFIG"),
+        "OBJCOPY=" + self.get_tool("OBJCOPY"),
+        "READELF=" + self.get_tool("READELF"),
+        "RANLIB=" + self.get_tool("RANLIB"),
+        "CXX=" + self.get_tool("CXX"),
+        "CPP=" + self.get_tool("CPP"),
+        "CC=" + self.get_tool("CC"),
+        "LD=" + self.get_tool("LD"),
+        "AR=" + self.get_tool("AR"),
+        "AS=" + self.get_tool("AS"),
+        "NM=" + self.get_tool("NM"),
+        "EXTRA_CFLAGS=" + self.get_cflags(shell=True),
+        "EXTRA_LDFLAGS=" + self.get_ldflags(shell=True),
+        "EXTRA_CXXFLAGS=" + self.get_cxxflags(shell=True),
+    ]
+
+    if self.stage > 0:
+        tool_args.append("OBJDUMP=" + self.get_tool("OBJDUMP"))
+
+    self.make.build(tool_args)
```

Comment:
delete this entire thing and set make_use_env = True and then set `self.make_build_args += extra_` inside init_build

Reply:
(as in, `self.make_build_args += [f"EXTRA_CFLAGS={self.get_cflags(shell=True)}", ...]`, etc)